java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBFileInfoStore.java

```diff
@@ -108,7 +109,7 @@ public void addFile(FileInfo fileInfo) throws StateStoreException {
                 || null == fileInfo.getPartitionId()) {
             throw new IllegalArgumentException("FileInfo needs non-null filename, status and partition: got " + fileInfo);
         }
-        Map<String, AttributeValue> itemValues = fileInfoFormat.createRecord(fileInfo);
+        Map<String, AttributeValue> itemValues = fileInfoFormat.createRecord(setLastUpdateTime(fileInfo));
```

Comment:
I think when we're setting the last update time on multiple files we should set them all to the same time, which should be created from `clock` immediately before we update the time on the first one. That way all the entries written to Dynamo will have the same update time if they were updated in the same transaction, which feels more representative of what's going on.